### DISCUSSION TITLE
Parsing tests freezes Neovim and uses a lot of memory (sometimes crashes)

### DISCUSSION THREAD

**Discussion by jfpedroza** [nvim-neotest/neotest#98] [Q&A] [answered]
I started working a Neotest adapter for Elixir [here](https://github.com/jfpedroza/neotest-elixir). I added parsing of tests and when I run `require("neotest").run.run()` or `require("neotest").summary.open()`, Neovim freezes and starts using more and more memory, up to about 4.2 GB.

When it doesn't crash, I can see the test tree. When I run it using stable (0.7.2), I can see the tree in just a second.

I tried using the `minimal.lua` from the issue template, but it works properly (even takes less than stable).

Any pointers to reproduce the issue would be appreciated.

**Comment by rcarriga** [comment, marked as answer]
Great to hear you're working on an adapter!

> When it doesn't crash, I can see the test tree. When I run it using stable (0.7.2), I can see the tree in just a second.

Interesting, so you're saying that with the exact same setup that stable works but nightly results in a crash? If so, have you tried a newer nightly?

Since you can't reproduce with the minimal init file, the only thing would be to bisect your config until you find the issue. The log file may have some useful information but can't guarantee.

**Reply by jfpedroza** [reply]
Yes, same setup, just switching between stable and nightly. I tried with latest nightly and still happens.

Yes, I need to bisect my config, but I haven't found the time for it, so for now I'm making progress using stable.

**Reply by rcarriga** [reply]
Can you provide a sample repo that causes the crash for you? I can try and reproduce myself

**Reply by jfpedroza** [reply]
I unsuccessfully tried to find what was causing the issue 😢. It's quite weird.

I removed every file in `after/plugin` and ran Neovim with `nvim -u NORC` (so, only the plugins) and could reproduce the issue.

Then I took the `minimal.lua` file and added every plugin and couldn't reproduce it that way.

My config is in https://github.com/jfpedroza/dotfiles/tree/master/nvim, if you want to give it a try.

**Reply by jfpedroza** [reply]
Sooo, this was fixed in the last two days. Not sure if it was a change in Neovim nightly or Neotest, but it now behaves properly.

**Reply by rcarriga** [reply]
Interesting, there were some performance improvements to the the treesitter parsing. I wonder if it had any effect. Can you try passing `fast = false` to the options table for `parse_positions` here https://github.com/jfpedroza/neotest-elixir/blob/master/lua/neotest-elixir/init.lua#L104 and see if it makes a difference?

**Reply by jfpedroza** [reply]
Indeed, setting `fast = false` returns the problem.

**Reply by rcarriga** [reply]
Ah OK! So what changed is that internally neovim by default, loads injections for the every treesitter tree. These are not needed for neotest and they slowed down the parsing because they're files that need to be read, so I've disabled them being read. I'm guessing that the injections for elixir caused the parser to perform poorly in some way. Either way great to hear it's fixed!
